scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/SCIMComplianceTest.java

```diff
@@ -70,11 +72,29 @@
 
   protected SCIMComplianceTest() {
     SCIMClientService scimClientService = configureScimClientService(SERVICE_URL);
-    userRequest = scimClientService.buildUserRequest();
-    groupRequest = scimClientService.buildGroupRequest();
-    schemaRequest = scimClientService.buildSchemaRequest();
+    SCIMRequest.Builder builder = SCIMRequest.newBuilder();
+    populateCustomHeaders(builder);
+
+    userRequest = scimClientService.buildUserRequest(builder);
+    groupRequest = scimClientService.buildGroupRequest(builder);
+    schemaRequest = scimClientService.buildSchemaRequest(builder);
   }
 
+  private void populateCustomHeaders(SCIMRequest.Builder newBuilder) {
+    if (HEADERS == null) {
+      return;
+    }
+
+    String[] customHeaders = HEADERS.split(",");
+    for (int i = 0; i < customHeaders.length; i++) {
+      int nextIndex = i + 1;
+      if (customHeaders.length > nextIndex) {
+        newBuilder.addHeader(customHeaders[i], customHeaders[nextIndex]);
+      }
+    }
+  }
```

Comment:
Can you use the for each loop instead to dealing with indexes

Reply:
maybe with reduce method could be improved like:
```
String[] customHeaders = HEADERS.split(",");
    Arrays.stream(customHeaders).reduce((s, s2) -> {
      newBuilder.addHeader(s, s2);
      return s;
    });
```